### ISSUE TITLE
[BUG] Dragging from a parent document into an iframe does not publish 

### ISSUE THREAD

**Issue by alexreardon** [microsoft/playwright#29278]
When performing a drag operation from a parent document to an iframe using `.dragTo()` in firefox, the iframe does not receive any drag events (eg "dragenter", "dragover" etc). 

- Events are correctly published for Chromium and Safari
- For Firefox, the parent document does receive the correct drag events when dragging from a child iframe to a parent document
- I can confirm that this is not a problem with firefox itself (I have tested). It looks like the Playwright adapter for firefox is not publishing any drag events during the drag operation in the iframe.

### System info
- Playwright Version: [1.30.0]
- Operating System: [macOS@14.2.1]
- Browser: [Firefox]
- Other info:

### Source code

Providing a full reproduction case is difficult here as it involves a server, but I can look into it if you want it.

```ts
import { expect, Page, test } from '@playwright/test';

async function setup({ page }: { page: Page }) {
  page.goto('${exampleUrl}');

  const dropTargetInParent = page.locator(
    '[data-testid="drop-target-in-parent"]',
  );
  const draggableInParent = page.locator('[data-testid="draggable-in-parent"]');

  const iframe = page.frameLocator('[data-testid="child-iframe"]');

  const dropTargetInIframe = iframe.locator(
    '[data-testid="drop-target-in-iframe"]',
  );
  const draggableInIframe = iframe.locator(
    '[data-testid="draggable-in-iframe"]',
  );

  const locators = {
    dropTargetInParent,
    draggableInParent,
    dropTargetInIframe,
    draggableInIframe,
  };

  // wait for everything to be ready before starting
  await Promise.all(
    Object.values(locators).map(locator =>
      locator.waitFor({ state: 'visible' }),
    ),
  );

  // asserting there is no drag data yet
  await expect(dropTargetInParent).toContainText('Latest drop data: none');
  await expect(dropTargetInIframe).toContainText('Latest drop data: none');

  return locators;
}

// broken for firefox
test('parent → iframe', async ({ page, browserName }) => {
  const { dropTargetInParent, dropTargetInIframe, draggableInParent } =
    await setup({ page });

  await draggableInParent.dragTo(dropTargetInIframe);

  await expect(dropTargetInParent).toContainText('Latest drop data: none');
  await expect(dropTargetInIframe).toContainText(
    'Latest drop data: Drag from parent: 0',
  );
});

// works for firefox
test('iframe → parent', async ({ page }) => {
  const { dropTargetInParent, dropTargetInIframe, draggableInIframe } =
    await setup({ page });

  await draggableInIframe.dragTo(dropTargetInParent);

  await expect(dropTargetInIframe).toContainText('Latest drop data: none');
  await expect(dropTargetInParent).toContainText(
    'Latest drop data: Drag from iframe: 0',
  );
});
```

`playwright.config.ts`

```ts
import { defineConfig, devices } from '@playwright/test';

export default defineConfig({
  reporter: 'html',
  projects: [
    {
      name: 'firefox',
      use: { ...devices['Desktop Firefox'] },
    },
  ],
});
```

**Steps**

- perform a drag operation (`.dragTo()`) from an element in a parent document to an inner iframe
- no drag events are published to the iframe (only pointer events)
- `dragTo()` completes successfully (according to Playwright), but not drag events where fired

**Expected**

Drag events are published correctly to firefox when dragging from a parent document to a child iframe

**Actual**

Drag events are not published when dragging over the child iframe in firefox

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your contribution to our project. This issue has been closed due to its limited upvotes and recent activity, and insufficient feedback for us to effectively act upon. Our priority is to focus on bugs that reflect higher user engagement and have actionable feedback, to ensure our bug database stays manageable.

Should you feel this closure was in error, please create a new issue and reference this one. We're open to revisiting it given increased support or additional clarity. Your understanding and cooperation are greatly appreciated.
